src/JsonSchema/SchemaStorage.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace JsonSchema;
+
+use JsonSchema\Entity\JsonPointer;
+use JsonSchema\Exception\UnresolvableJsonPointerException;
+use JsonSchema\Iterator\ObjectIterator;
+use JsonSchema\Uri\UriResolver;
+use JsonSchema\Uri\UriRetriever;
+
+class SchemaStorage
+{
+    protected $uriRetriever;
+    protected $uriResolver;
+    protected $schemas = array();
+
+    public function __construct(
+        UriRetrieverInterface $uriRetriever = null,
+        UriResolverInterface $uriResolver = null
+    )
+    {
+        $this->uriRetriever = $uriRetriever ?: new UriRetriever;
+        $this->uriResolver = $uriResolver ?: new UriResolver;
+    }
+
+    /**
+     * @return UriRetrieverInterface
+     */
+    public function getUriRetriever()
+    {
+        return $this->uriRetriever;
+    }
+
+    /**
+     * @return UriResolverInterface
+     */
+    public function getUriResolver()
+    {
+        return $this->uriResolver;
+    }
+
+    public function addSchema($id, $schema = null)
```

Comment:
`$id`? I'm not sure, if this is as informative as it could be. The parameter name should indicate what the parameter represents. Actually I would think about a unique identifier, like a number/integer. But if I understand it correctly, it's an Uri, that resolves to a json schema. So probably you want to change the parameter name.

Also: missing phpdoc block.